source/entities/Building.cpp

```diff
@@ -115,11 +115,19 @@ bool Building::canBuildingBeRemoved()
     bool ret = true;
     for(std::pair<Tile* const, TileData*>& p : mTileData)
     {
-        if(!p.second->mSeatsVision.empty())
+        for(Seat* seat : p.second->mSeatsVision)
         {
+            if(seat->getPlayer() == nullptr)
+                continue;
+            if(!seat->getPlayer()->getIsHuman())
+                continue;
+
             ret = false;
             break;
```

Comment:
no, it's not the same for loop. The idea is that we want to check if a human player has seen the room when built and not since it is destroyed. As long as it is the case, we do not remove it.